config/routes.rb

```diff
@@ -701,6 +701,14 @@
     end
   end
 
+   ### SINGLE PAGE
+
+  resources :single_pages do
+    member do
+      get "/render_item_detail/:id/type/:type" => "single_pages#render_item_detail"
```

Comment:
I still think it would be easier if you did this through each type's controller instead of a single controller.

You would not have to check types and could use the existing methods to find and authorize assets.

Reply:
Doing this through each type's controller works but it's not possible to set `@single_page` in order to use `displaying_single_page` helper method in the views. Is there any solution for this?

Reply:
Why not? Can't you just set it in the same way?

Reply:
I can set it as it is. I tried to implement your approach but, I noticed that the type controller gets active when for example send this Ajax request: `/investigations/1` (then it shows `@single_page` as undefined)

Reply:
You would need to call `/investigations/1/render_item_detail` and have a `before_action ... only: :render_item_detail`.

We have a similar feature for "previews", which is available for all asset + ISA types which you might want to look at*

https://github.com/seek4science/seek/blob/master/lib/seek/preview_handling.rb
https://github.com/seek4science/seek/blob/master/lib/seek/assets_common.rb#L7
https://github.com/seek4science/seek/blob/master/config/routes.rb#L94
https://github.com/seek4science/seek/blob/master/config/routes.rb#L108

*Although the behaviour is slightly wrong/weird since the route ends up like `/data_files/preview?id=123`, and it does the authorization in the view.